src/meta_ops.rs

```diff
@@ -117,26 +218,28 @@ pub fn index<'gc>(
                 let table = stack.get(0);
                 let key = stack.get(1);
                 stack.clear();
+                // TODO: detect index chain length, error if too long
+                // Example: t = {}; setmetatable(t, { __index = t }); t.a
+                // TODO: note chain depth in other errors?
```

Comment:
We should not do this, there's no reason to. You can write an infinite loop already in plain Lua, too.